src/sbvr-api/permissions.ts

```diff
@@ -684,6 +685,201 @@ const getAlias = (name: string) => {
 	}
 	return `permissions${permissionsJSON}`;
 };
+
+const rewriteRelationship = memoizeWeak(
+	(
+		value: Relationship,
+		name: string,
+		abstractSqlModel: AbstractSqlModel,
+		permissionsLookup: PermissionLookup,
+		vocabulary: string,
+	) => {
+		let escapedName = sqlNameToODataName(name);
+		if (abstractSqlModel.tables[name]) {
+			escapedName = sqlNameToODataName(abstractSqlModel.tables[name].name);
+		}
+
+		const originalAbstractSQLModel = sbvrUtils.getAbstractSqlModel({
+			vocabulary,
+		});
+
+		const rewrite = (object: Relationship | RelationshipMapping) => {
+			if ('$' in object && Array.isArray(object.$)) {
+				// object is in the form of
+				// { "$": ["actor", ["actor", "id"]] } or { "$": ["device type"] }
+				// we are only interested in the first case, since this is a relationship
+				// to a different resource
+				const mapping = object.$;
+				if (
+					mapping.length === 2 &&
+					Array.isArray(mapping[1]) &&
+					mapping[1].length === 2 &&
+					typeof mapping[1][0] === 'string'
+				) {
+					// now have ensured that mapping looks like ["actor", ["actor", "id"]]
+					// this relations ship means that:
+					//   mapping[0] is the local field
+					//   mapping[1] is the reference to the other resource, that joins this resource
+					//   mapping[1][0] is the name of the other resource (actor in the example)
+					//   mapping[1][1] is the name of the field on the other resource
+					//
+					// this therefore defines that the local field `actor` needs
+					// to match the `id` of the `actor` resources for the join
+
+					const possibleTargetResourceName = mapping[1][0];
+
+					// Skip this if we already shortcut this connection
+					if (possibleTargetResourceName.endsWith('$bypass')) {
+						return;
+					}
+
+					const targetResource =
+						abstractSqlModel.tables[possibleTargetResourceName];
+					const targetResourceEscaped = sqlNameToODataName(
+						targetResource?.name ?? possibleTargetResourceName,
+					);
```

Comment:
I don't think `targetResource` is used anywhere else right?
```suggestion
					const targetResourceEscaped = sqlNameToODataName(
						abstractSqlModel.tables[possibleTargetResourceName]?.name ?? possibleTargetResourceName,
					);
```